### PR TITLE
hosts routes and button show only if user is admin

### LINKED ISSUE #301: hosts only visible for admins
[//]: # (Lines starting with [//]: are considered to be comments. You do not have to delete those lines, they are not rendered upon release.)

**User Story**

As an admin, only I need to see the "Hosts" page, as well as the button in the header.

**Acceptance Criteria:**

- [ ] The button "hosts" is only visible to admins
- [ ] The page /hosts is only available to admins

[//]: # (Please don't forget to add the priority and complexity label to this issue.)

### PR DESCRIPTION
**Pull request by skydivin4ng3l**
Fixes #301

### CODE REVIEW

File: spec/controllers/hosts_controller_spec.rb [hunk 1]
```diff
@@ -4,21 +4,36 @@
...
+      it 'renders index page' do
+        expect(get(:index)).to render_template('hosts/index')
+      end
     end
 
-    it 'renders index page' do
-      expect(get(:index)).to render_template('hosts/index')
+    context 'current user is not an admin' do
+      let(:current_user) { FactoryBot.create :user }
+
```
Comment (lassesteffen) [inline spec/controllers/hosts_controller_spec.rb]: you are missing a before do block

before do
get(;index)
end

File: spec/controllers/hosts_controller_spec.rb [hunk 2]
```diff
@@ -30,27 +45,42 @@
...
+        expect(get(:show, params: { id: 1 })).to render_template('errors/not_found')
+      end
     end
 
-    it 'renders not found page when no host found' do
-      allow(double_api).to receive(:get_host).and_return(nil)
-      expect(get(:show, params: { id: 1 })).to render_template('errors/not_found')
+    context 'current user is not an admin' do
+      let(:current_user) { FactoryBot.create :user }
+
```
Comment (lassesteffen) [inline spec/controllers/hosts_controller_spec.rb]: you are missing a before do block

before do
get(:index)
end

**Comment by lassesteffen** [comment]
Implementing my comments should make the specs pass. They are correct, you are just not making a request

**Comment by skydivin4ng3l** [comment]
> Implementing my comments should make the specs pass. They are correct, you are just not making a request

Interesting, then the tests for hosts never worked correctly?

**Comment by lassesteffen** [comment]
No, they worked previously. You just need to make sure you make the request you test for in every test case. They did it in each it-Statement, im proposing to put in a before do loop. But in the end it just needs to happen somewhere.

**Comment by skydivin4ng3l** [comment]
Still failing

Expected HostsController to set flash.now[:alert] to "You do not have the necessary rights to perform this action.", but it did not

  0) HostsController GET #index current user is not an admin notifies about missing rights
     Failure/Error: should set_flash.now[:alert].to('You do not have the necessary rights to perform this action.')
       Expected HostsController to set flash.now[:alert] to "You do not have the necessary rights to perform this action.", but it did not
     # ./spec/controllers/hosts_controller_spec.rb:39:in `block (4 levels) in <top (required)>'

Expected HostsController to set flash.now[:alert] to "You do not have the necessary rights to perform this action.", but it did not

  0) HostsController get #show current user is not an admin notifies about missing rights
     Failure/Error: should set_flash.now[:alert].to('You do not have the necessary rights to perform this action.')
       Expected HostsController to set flash.now[:alert] to "You do not have the necessary rights to perform this action.", but it did not
     # ./spec/controllers/hosts_controller_spec.rb:90:in `block (4 levels) in <top (required)>'

**Comment by lassesteffen** [comment]
I am looking into it, not sure how to solve this problem right now. Did you try it out in the browser to see if it flashed the error message for you?

File: spec/controllers/hosts_controller_spec.rb [hunk 3]
```diff
@@ -49,46 +42,51 @@
...
+          expect(response).to render_template('errors/not_found')
+        end
       end
     end
 
-    context 'current user is not an admin' do
+    context 'when the current user is not an admin' do
       let(:current_user) { FactoryBot.create :user }
 
-      before :each do
```
[the remark this answers is not in the document]
Comment (lassesteffen) [reply spec/controllers/hosts_controller_spec.rb]: Because it is the default value. If you do not specify anything it is going to be :each

**Comment by BjoernDaase** [comment]
> > The PR is good. Please check the issue if it is intended behavious that also hosts/specific hostpage is only visible for admins!
> 
> @isabel-amon this is the intended behavior isn't it? it is stated so in the issue, please correct if wrong.

Yes, I can confirm that that's the intended behavior.

**Pull request merged**